core/clustersmngr/factory.go

```diff
@@ -104,7 +135,28 @@ func NewClientFactory(fetcher ClusterFetcher, nsChecker nsaccess.Checker, logger
 		initialClustersLoad: make(chan bool),
 		scheme:              scheme,
 		newClustersPool:     clusterPoolFactory,
+		watchers:            []*ClustersWatcher{},
+	}
+}
+
+// Subscribe returns a new ClustersWatcher.
+func (cf *clientsFactory) Subscribe() *ClustersWatcher {
+	cw := &ClustersWatcher{cf: cf, Updates: make(chan ClusterListUpdate, 5)}
+	cf.watchers = append(cf.watchers, cw)
+
+	return cw
+}
+
+// RemoveWatcher removes the given ClustersWatcher from the list of watchers.
+func (cf *clientsFactory) RemoveWatcher(cw *ClustersWatcher) {
+	watchers := []*ClustersWatcher{}
+	for _, w := range cf.watchers {
+		if cw != w {
```

Comment:
We are only comparing the channel's value here. Is that enough or we can maybe generate an id for each watcher?

Reply:
We're comparing the `ClustersWatcher` not the channel.

Reply:
Comparing structs is done through the struct fields and since `cf` will be the same for all watcher in a cluster factory, that's what we are essentially checking on or am I missing something?

Reply:
@sarataha do you wanna write maybe one more test :)

Where we add 2 watchers, remove 1, update, make sure the one that is still subscribed gets the update?

Reply:
> Comparing structs is done through the struct fields

So I think we're comparing pointers here. They only get auto-dereferenced if you use a _selector_, https://go.dev/ref/spec#Selectors